### PR TITLE
Double & triple shot

### PR DESCRIPTION
**Pull request by plopyy**
Hi !

I thought about special effects for bows and magic staves, i make this pull request.
I thought about special effects for bows and magic staves.
So I created MultiShot.cs to handle the two new MagicEffectType (DoubleMagicShot & TripleBowShot)
I put them "NoRoll": true
to put them on unique legendary weapons or as set bonuses

### CODE REVIEW

File: EpicLoot/MagicItemEffects/MultiShot.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{
+    [HarmonyPatch]
+    public static class MultiShot
+    {
+        [HarmonyPatch(typeof(Attack), "FireProjectileBurst")]
+        class FireProjectileBurstPatch
+        {
+            public static void Prefix(Attack __instance)
+            {
+                var player = (Player)__instance.m_character;
```
Comment (RandyKnapp) [inline EpicLoot/MagicItemEffects/MultiShot.cs:19]: I would do a null check here on __instance and early out if null. Remember to do proper null comparisons (object == null) for Unity objects.

File: EpicLoot/MagicItemEffects/MultiShot.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                var weaponDamage = __instance.GetWeapon()?.GetDamage();
+                var ItemType = __instance.GetWeapon()?.m_shared.m_itemType;
+                var SkillType = __instance.GetWeapon()?.m_shared.m_skillType;
+
+                if (player.HasActiveMagicEffect(MagicEffectType.DoubleMagicShot) && weaponDamage.HasValue && ItemType == ItemDrop.ItemData.ItemType.TwoHandedWeapon && SkillType == Skills.SkillType.ElementalMagic)
```
Comment (RandyKnapp) [inline EpicLoot/MagicItemEffects/MultiShot.cs:24]: If any of the above are null, you should return before using them

**Review by RandyKnapp — CHANGES_REQUESTED** [review]
Hey this looks like a great addition to the mod, thank you! I have a couple of changes requested for your main effect method. Thank you!

**Review by RandyKnapp — APPROVED** [review]

**Comment by RandyKnapp** [comment]
Soon (TM)

**Comment by RandyKnapp** [comment]
Like, I appreciate your help with this, but I will need to personally test it and ensure it's ready for release. It will take a little time.

**Comment by plopyy** [comment]
I tested it to check if it work, that was fun :D, and thanks for quick reply n_n !

**Review by RandyKnapp — APPROVED** [review]

**Pull request merged**